internal/controller/daemonset/instaslice_daemonset.go

```diff
@@ -363,40 +326,59 @@ func (r *InstaSliceDaemonsetReconciler) SetupWithManager(mgr ctrl.Manager) error
 			Name:      r.NodeName,
 			Namespace: controller.InstaSliceOperatorNamespace,
 		}
-		errRetrievingInstaSliceForSetup := r.Get(ctx, typeNamespacedName, &instaslice)
-		if errRetrievingInstaSliceForSetup != nil {
-			log.Error(errRetrievingInstaSliceForSetup, "unable to fetch InstaSlice resource for node")
+		err := r.Get(ctx, typeNamespacedName, &instaslice)
+		if err != nil {
+			log.Error(err, "Unable to fetch Instaslice for node", "nodeName", r.NodeName)
 		}
 
 		if r.Config.EmulatorModeEnable {
-			err := r.Create(ctx, utils.GenerateFakeCapacity(r.NodeName))
+			fakeCapacity := utils.GenerateFakeCapacity(r.NodeName)
+			err := r.Create(ctx, fakeCapacity)
 			if err != nil && !errors.IsAlreadyExists(err) {
 				log.Error(err, "could not create fake capacity", "node_name", r.NodeName)
 				return err
 			}
+			//get metadata.resourceVersion
+			for i := 1; i < 5; i++ {
+				err := r.Get(ctx, typeNamespacedName, &instaslice)
+				if err == nil {
+					break
+				}
+				log.Error(err, "Retrying fetch fake capacity", "attempt", i+1, "node_name", r.NodeName)
+				time.Sleep(2 * time.Second)
```

Comment:
The sleeps here are strange.

Use something from k8s.io/apimachinery/pkg/util/wait ?